ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-v4-simulation/value-display/value-display.tsx

```diff
@@ -88,18 +94,25 @@ const PermitSimulationValueDisplay: React.FC<
     return undefined;
   }, [exchangeRate, tokenDecimals, value]);
 
-  const { tokenValue, tokenValueMaxPrecision } = useMemo(() => {
-    if (!value || tokenId) {
-      return { tokenValue: null, tokenValueMaxPrecision: null };
-    }
+  const { tokenValue, tokenValueMaxPrecision, shouldShowUnlimitedValue } =
+    useMemo(() => {
+      if (!value || tokenId) {
+        return {
+          tokenValue: null,
+          tokenValueMaxPrecision: null,
+          shouldShowUnlimitedValue: false,
+        };
+      }
 
-    const tokenAmount = calcTokenAmount(value, tokenDecimals);
+      const tokenAmount = calcTokenAmount(value, tokenDecimals);
 
-    return {
-      tokenValue: formatAmount('en-US', tokenAmount),
-      tokenValueMaxPrecision: formatAmountMaxPrecision('en-US', tokenAmount),
-    };
-  }, [tokenDecimals, value]);
+      return {
+        tokenValue: formatAmount('en-US', tokenAmount),
+        tokenValueMaxPrecision: formatAmountMaxPrecision('en-US', tokenAmount),
+        shouldShowUnlimitedValue:
+          canDisplayValueAsUnlimited && Number(value) > UNLIMITED_THRESHOLD,
```

Comment:
Why do we have to cast, what other type could it be?

Reply:
value can be both a number or a string